Moar filter refactoring

dnahodil opened the pull request:
Some tidy-up in LayerController from Filters work

danfruehauf commented on grails-app/controllers/au/org/emii/portal/LayerController.groovy:
```diff
@@ -25,6 +25,12 @@ class LayerController {
 
     static allowedMethods = [save: "POST", update: "POST", delete: "POST"]
 
+    static final HTTP_200_OK = 200
```
Perhaps extract this to a more common file? I bet we can use it in many more places...

dnahodil replied:
I'll have a look at expanding this to our other controllers

danfruehauf commented on grails-app/controllers/au/org/emii/portal/LayerController.groovy:
```diff
@@ -642,45 +647,52 @@ class LayerController {
-            def serverObject = _getServerClass(params.serverType)
 
-            render text: serverObject.getStyles(server, layer) as JSON
+        def (server, layer, serverType) = parseParams(params)
```
That's slick. I didn't know grails does that for you.

dnahodil replied:
Yeah, it's a neat trick.

dnahodil commented:
I've updated all our controllers to use the HTTP status constants.

I changed all the controllers despite the fact we'll lose a lot of them to no_db changes. It was just easier doing them all.

danfruehauf commented:
Yep, that looks good.
